### ISSUE TITLE
[Meta] Planned changes to the manifest format

### ISSUE THREAD

**Issue by ruflin** [elastic/package-registry#517]
A few changes are planned to the manifest(s) format. This is to track these changes:

# Rename `datasources` to `config_templates` 

We remove the term datasources across the different components. This now also has to be removed in the packages and be renamed to config_templates:

  * [x] Update Kibana https://github.com/elastic/kibana/pull/69864
  * [x] Update registry https://github.com/elastic/package-registry/pull/514  https://github.com/elastic/package-registry/pull/570
  * [x] Packages https://github.com/elastic/package-storage/pull/94
  * [x] Integrations https://github.com/elastic/integrations/pull/132

# Cleanup terms

Cleanup terms used inside the registry.
* [x] Cleanup dataset terms https://github.com/elastic/package-registry/pull/514

# Requirements to constraints

We currently use requirements section. To align with the agent and other parts, it should be renamed to constraints. In line with this task, also the default dependencies in the packages should be udpated.

* [x] Update Registry  https://github.com/elastic/package-registry/pull/519
* [x] Update Integrations https://github.com/elastic/integrations/pull/133
* [x] Update Package storage  https://github.com/elastic/package-storage/pull/95
* [x] Remove legacy part in registry https://github.com/elastic/package-registry/pull/580
* [x] Introduce new default for all packages (^7.9.0) https://github.com/elastic/integrations/issues/151

# Update categories

* [x] Implement new categories in package-registry https://github.com/elastic/package-registry/pull/534
* [x] Remove existing categories in package-storage https://github.com/elastic/package-storage/pull/97
* [x] Update packages with new categories in integrations and release packages https://github.com/elastic/integrations/issues/135
* [x] Remove legacy package information in registry https://github.com/elastic/package-registry/pull/571

# Directory names cleanup

Some of the directory names contained `-`, others `_`. This also had an effect on how the config options are called and what paths users have to define. To only have one way, it was decided to use `_` everywhere and packages need to be adjusted.

* [x] Update directory names to be _ instead of - https://github.com/elastic/package-registry/issues/565
* [x] Update packages to also use the new name https://github.com/elastic/package-storage/pull/90
* [x] Change Kibana https://github.com/elastic/kibana/pull/70320
* [x] Update packages to only use the new names
* [x] Update integrations repo https://github.com/elastic/integrations/issues/150
* [ ] Validate all paths to not contain `-` if directory. https://github.com/elastic/package-registry/pull/572

# Ingest pipeline config option

The ingest pipeline config option should be moved under the new Elasticsearch object for consistency

* [x] Add new option in registry https://github.com/elastic/package-registry/pull/564
* [ ] Add support for the option in Kibana
* [ ] Modify packages to only use the new option
* [ ]  Remove legacy implementation in the registry

# Rename input logs to logfile

* [x] Add support on the Agent side
* [x] Change it in integrations https://github.com/elastic/integrations/pull/146
* [x] Change it in package-storage https://github.com/elastic/package-storage/pull/126
* [x] Remove support for `logs` on the agent side

# More

* [x] Remove OS specifics https://github.com/elastic/package-registry/pull/549

More details to come

**Comment by mtojek** [comment]
Not sure if manifest files should change so much in time. I wonder what is the root cause of such evolving? Is it a matter of a discussion?

Also, what about existing packages (I'm asking specifically about `Rename datasources to config_templates`)?

**Comment by ruflin** [comment]
The main reason for the changes are that we discussed quite a few naming changes over the last weeks in Agent / Kibana which I think should also make it into the packages to be consistent. And 7.9 is our last chance to really break things, so I rather do them all now. Happy to share more details related to each change.

For the `datasources` to `config_templates` I would expect us to "break" existing packages again.  A general thought I have for 7.9 is that shortly before the release, we archive all the older versions of the packages we pushed so far as these will never be installed anyways.
